cni/network/network.go

```diff
@@ -912,7 +911,7 @@ func (plugin *netPlugin) Delete(args *cniSkel.CmdArgs) error {
 	if err != nil {
 		log.Printf("[cni-net] Failed to extract network name from network config. error: %v", err)
 
-		if !utils.IsNotFoundError(err) {
+		if !cnsclient.IsNotFound(err) {
```

Comment:
>cnsclient [](http://example.com/codeflow?start=6&length=9)

Not every scenario has cnsClient. It is not used for AKS single tenancy.

Reply:
the right way is to have shim (one more layer) here for getnetworkname

Reply:
there is no further dependency taken here (see imports). the check is appropriate because the only error handled seems to have always been a CNS specific error.

Reply:
cnsclient package is specific to swift singletenancy and this function looks like common for both swift and traditional. if so, then it should be moved to common package imo.

Reply:
broad common or util packages are an antipattern in go, this function deals only with the `CNSClientError` so it should live in the same file as the `CNSClientError` (which should live next to the `CNSClient` since that's the only thing that returns it).
that cnsclient is a poor choice of package name, and that that client is initialized as a global with package side-effects are both bad things we can argue, but are out of scope on this change.

Reply:
i went through `getnetworkname` function and looks like it returns error only in windows multitenancy scenario. we can ignore this error in getNetworkName function itself instead of checking here.. also i found even if we ignore error at this step, it will still fail in `if nwInfo, err = plugin.nm.GetNetworkInfo(networkId); err != nil {` and return without any useful action. may be its not scoped to this PR, but we should invoke cns here to cleanup based on containerid as like traditional case.

Reply:
i will take back my comment on checking in getNetworkName..that doesn't sound right but can we add a check in isnotfound function to do this only in case of multitenancy

Reply:
I think it's safer to just check the error type regardless of the scenario. No need to do a scenario check to try to avoid a single if statement that will evaluate to false. If we started doing scenario checks, the code will get really complicated when we add more and more scenarios.